src/App.tsx

```diff
@@ -1,61 +1,129 @@
+import { useState } from 'react';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+function getUserByName(userName: string): User | undefined {
+  return usersFromServer.find(user => user.name === userName);
+}
 
 export const App = () => {
+  const [visibleTodos, setTodos] = useState(todos);
+  const [author, setAuthor] = useState('Choose a user');
+  const [task, setTask] = useState('');
+  const [isActiveTask, setIsActiveTask] = useState(false);
+  const [isActiveAuthor, setIsActiveAuthor] = useState(false);
+  const [isTaskError, setIsTaskError] = useState(false);
+  const [isAuthorError, setIsAuthorError] = useState(false);
+
+  const updateTodos = () => {
+    setIsTaskError(!isActiveTask);
+    setIsAuthorError(!isActiveAuthor);
+
+    if (isActiveTask && isActiveAuthor) {
+      const maxId = Math.max(...todos.map(todo => todo.id));
+      const newTodo: Todo = {
+        id: maxId + 1,
+        userId: getUserByName(author)?.id,
+        title: task,
+        completed: false,
+        user: getUserByName(author),
+      };
+
+      todos.push(newTodo);
+      setTodos(todos);
+      setTask('');
+      setAuthor('Choose a user');
+      setIsActiveTask(false);
+      setIsActiveAuthor(false);
+      setIsTaskError(false);
+      setIsAuthorError(false);
+    }
+  };
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/users" method="POST">
+      <form
+        action="/api/users"
+        method="POST"
+        onSubmit={(e) => {
+          e.preventDefault();
+          updateTodos();
+        }}
+      >
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <label htmlFor="title">Title: </label>
+
+          <input
+            type="text"
+            data-cy="titleInput"
+            id="title"
+            placeholder="Enter a title"
+            value={task}
+            onChange={(e) => {
+              setTask(e.target.value);
+              setIsActiveTask(true);
+              setIsTaskError(false);
+            }}
+          />
+          {isTaskError
+          && <span className="error">Please enter a title</span>}
         </div>
 
         <div className="field">
-          <select data-cy="userSelect">
-            <option value="0" disabled>Choose a user</option>
-          </select>
+          <label htmlFor="title">User: </label>
```

Comment:
It should be a label for the user, not for the title
![Animation](https://user-images.githubusercontent.com/99403198/198831012-189ae896-615e-4122-9606-42065e2ae149.gif)